lib/factory_bot/definition.rb

```diff
@@ -191,5 +193,14 @@ def automatically_register_defined_enums?(klass)
       FactoryBot.automatically_define_enum_traits &&
         klass.respond_to?(:defined_enums)
     end
+
+    def ensure_attribute_not_generating_primary_key!(attribute, for_class:)
+      case attribute
+      when FactoryBot::Attribute::Sequence
+        if for_class < ActiveRecord::Base && for_class.primary_key == attribute.name.to_s
+          raise AttributeDefinitionError, "Attribute generates an Active Record primary key"
```

Comment:
Will the error message the user sees provide enough information to determine which factory and attribute needs to change? For example, would they be able to tell that their `SomeModel` factory's `some_id` property is the issue at hand?